File: Tutorial5-2Dagger-Android-SubComponents3-ContributesAndroidInjector/src/main/java/com/example/tutorial5_2dagger_android_subcomponents3/SecondActivity.java
package com.example.tutorial5_2dagger_android_subcomponents3;

import android.content.SharedPreferences;
import android.os.Bundle;

import androidx.annotation.Nullable;

import javax.inject.Inject;

import dagger.android.support.DaggerAppCompatActivity;

public class SecondActivity extends DaggerAppCompatActivity {

    // Injected from AppModule with @Singleton
    @Inject
    SharedPreferences sharedPreferences;

    // Injected from SecondActivity SubComponent Module with @ActivityScope
    @Inject
    ToastMaker toastMaker;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        System.out.println("👀 SecondActivity toastMaker: " + toastMaker);


        toastMaker.showToast("sharedPreferences " + sharedPreferences);
    }
}

File: Tutorial5-2Dagger-Android-SubComponents3-ContributesAndroidInjector/src/main/java/com/example/tutorial5_2dagger_android_subcomponents3/SecondActivity.kt
package com.example.tutorial5_2dagger_android_subcomponents3

import android.content.SharedPreferences
import android.os.Bundle
import com.example.tutorial5_2dagger_android_subcomponents2.model.ToastMaker
import dagger.android.support.DaggerAppCompatActivity
import javax.inject.Inject

class SecondActivity : DaggerAppCompatActivity() {

    // Injected from AppModule with @Singleton
    @Inject
    lateinit var sharedPreferences: SharedPreferences

    // Injected from SecondActivity SubComponent Module with @ActivityScope
    @Inject
    lateinit var toastMaker: ToastMaker

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        println("👀 SecondActivity toastMaker: $toastMaker")
        toastMaker.showToast("sharedPreferences $sharedPreferences")
    }
}